From 1159f5ecb2a75823bf25054c2e65076d7276d38a Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 28 Sep 2017 21:05:08 +0800
Subject: [PATCH 2/8] sunxi: update clocks for display on R40

The clocks for TCONs on R40 are different from other SoCs, as there's 4
TCONs on R40; in addition an extra bus gate is added for HDMI.

Update the clocks' offset for R40 display.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 arch/arm/include/asm/arch-sunxi/clock_sun6i.h | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm/include/asm/arch-sunxi/clock_sun6i.h b/arch/arm/include/asm/arch-sunxi/clock_sun6i.h
index d328df9597..b6d951c0fc 100644
--- a/arch/arm/include/asm/arch-sunxi/clock_sun6i.h
+++ b/arch/arm/include/asm/arch-sunxi/clock_sun6i.h
@@ -75,6 +75,7 @@ struct sunxi_ccm_reg {
 #endif
 	u32 be1_clk_cfg;	/* 0x108 BE1 module clock */
 	u32 fe0_clk_cfg;	/* 0x10c FE0 module clock */
+#ifndef CONFIG_MACH_SUN8I_R40
 	u32 fe1_clk_cfg;	/* 0x110 FE1 module clock */
 	u32 mp_clk_cfg;		/* 0x114 MP module clock */
 #ifdef CONFIG_SUNXI_DE2
@@ -84,6 +85,12 @@ struct sunxi_ccm_reg {
 	u32 lcd0_ch0_clk_cfg;	/* 0x118 LCD0 CH0 module clock */
 	u32 lcd1_ch0_clk_cfg;	/* 0x11c LCD1 CH0 module clock */
 #endif
+#else
+	u32 tcon_lcd0_clk_cfg;	/* 0x110 TCON0 LCD module clock */
+	u32 tcon_lcd1_clk_cfg;	/* 0x114 TCON1 LCD module clock */
+	u32 tcon_tv0_clk_cfg;	/* 0x118 TCON0 TV module clock */
+	u32 tcon_tv1_clk_cfg;	/* 0x11c TCON1 TV module clock */
+#endif
 	u32 tve_clk_cfg;	/* 0x120 H3/H5 TVE module clock */
 	u32 reserved14[2];
 	u32 lcd0_ch1_clk_cfg;	/* 0x12c LCD0 CH1 module clock */
@@ -303,11 +310,17 @@ struct sunxi_ccm_reg {
 #define AHB_GATE_OFFSET_SS		5
 
 /* ahb_gate1 offsets */
+#define AHB_GATE_OFFSET_TCON_TOP	30
+#define AHB_GATE_OFFSET_TCON_TV1	29
+#define AHB_GATE_OFFSET_TCON_TV0	28
+#define AHB_GATE_OFFSET_TCON_LCD1	27
+#define AHB_GATE_OFFSET_TCON_LCD0	26
 #define AHB_GATE_OFFSET_DRC0		25
 #define AHB_GATE_OFFSET_DE_FE0		14
 #define AHB_GATE_OFFSET_DE_BE0		12
 #define AHB_GATE_OFFSET_DE		12
 #define AHB_GATE_OFFSET_HDMI		11
+#define AHB_GATE_OFFSET_HDMI2		10
 #define AHB_GATE_OFFSET_TVE		9
 #ifndef CONFIG_SUNXI_DE2
 #define AHB_GATE_OFFSET_LCD1		5
@@ -446,6 +459,11 @@ struct sunxi_ccm_reg {
 #define AHB_RESET_OFFSET_SS		5
 
 /* ahb_reset1 offsets */
+#define AHB_RESET_OFFSET_TCON_TOP	30
+#define AHB_RESET_OFFSET_TCON_TV1	29
+#define AHB_RESET_OFFSET_TCON_TV0	28
+#define AHB_RESET_OFFSET_TCON_LCD1	27
+#define AHB_RESET_OFFSET_TCON_LCD0	26
 #define AHB_RESET_OFFSET_SAT		26
 #define AHB_RESET_OFFSET_DRC0		25
 #define AHB_RESET_OFFSET_DE_FE0		14
-- 
2.13.5

